### PR TITLE
Issue67

### PR DESCRIPTION
**Pull request by sbird**
Issue67 v 2.

### CODE REVIEW

**Comment by sbird** [comment]
This is Karime's issue 67 fix with a couple more changes on top. Notably save some more useful BH fields to disc.

File: libgadget/blackhole.c [hunk 1]
```diff
@@ -585,28 +592,22 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
...
-        /* We do not know how to notify the tree of mass changes. so
-         * blindly enforce a mass conservation for now. */
+        /* Leave the swallowed BH mass around
+         * so we can work out mass at merger. */
         O->Mass += (P[other].Mass);
         O->BH_Mass += (BHP(other).Mass);
-        P[other].Mass = 0;
-        BHP(other).Mass = 0;
-
-        slots_mark_garbage(other, PartManager, SlotsManager);
```
Comment (rainwoodman) [inline libgadget/blackhole.c]: Do we no longer mark the swallowed particle a  garbage?

Comment (sbird) [reply libgadget/blackhole.c]: That's right - we don't want to GC it as per #67 we want to keep the swallowed particle around for merger tree stuff.

Comment (sbird) [reply libgadget/blackhole.c]: This is actually a much better way of tracking mergers than the usual merger tree code - I'm surprised other people haven't done it. It kind of obsoletes an entire complicated subfield...

Comment (maamari) [reply libgadget/blackhole.c]: What is meant by obsoletes an entire subfield? Which subfield is that in reference to

Comment (sbird) [reply libgadget/blackhole.c]: There are a whole bunch of people who build algorithms to follow the mergers of dark matter halos from incomplete temporal sampling of snapshots, for example https://arxiv.org/abs/1307.3577

**Comment by rainwoodman** [comment]
Using "tracer particles" is not exactly a new idea though.

People do worry about the tracer gets ejected or stolen.

**Comment by sbird** [comment]
It's true, this is not novel really. But it does seem rarer than I would expect.

File: libgadget/blackhole.c (hunk 1 above)
Comment (maamari) [reply libgadget/blackhole.c]: Ah okay interesting, thanks for the link

**Comment by rainwoodman** [comment]
I guess it was because it was not in a public version of Gadget.

**Pull request merged**